File: frontend/src/stores/system.js
import { defineStore } from "pinia";
import { getSystemInfo } from "../api/client";

// 系统信息 store：缓存后端模型配置，避免每个页面重复请求。
export const useSystemStore = defineStore("system", {
  state: () => ({
    // 默认值用于首屏占位，接口返回后会被覆盖。
    info: {
      app_name: "Personal Scholar Agent API",
      model_provider: "LocalRuleEngine",
      llm_model_name: "DemoPipeline-v1",
      generation_model_name: "DeepSeek-V3",
      evaluation_model_name: "Qwen3",
      collaboration_mode: "Multi-Agent Collaboration: DeepSeek-V3 (Gen) + Qwen3 (Eval)",
      embedding_model_name: "TokenOverlapRetriever-v1",
      pipeline_mode: "DeterministicDraft"
    },
    loaded: false
  }),
  actions: {
    // 拉取系统信息：GET /api/system/info -> info 对象。
    async fetchInfo() {
      try {
        const { data } = await getSystemInfo();
        this.info = data;
      } finally {
        this.loaded = true;
      }
    }
  }
});

File: backend/app/routers/system.py
from fastapi import APIRouter

from ..config import get_settings
from ..dependencies import get_skill_registry
from ..schemas import SystemInfoResponse

router = APIRouter(tags=["system"])
settings = get_settings()

@router.get("/system/info", response_model=SystemInfoResponse)
async def get_system_info() -> SystemInfoResponse:
    use_deepseek = bool(settings.deepseek_api_key.strip())
    collaboration_mode = (
        f"Multi-Agent Collaboration: "
        f"{settings.generation_model_name} (Gen) + {settings.evaluation_model_name} (Eval)"
    )
    skill_registry = get_skill_registry()
    skill_status = skill_registry.status()
    return SystemInfoResponse(
        app_name=settings.app_name,
        model_provider="DeepSeek + DashScope(Qwen3)" if use_deepseek else settings.model_provider,
        llm_model_name=settings.deepseek_model if use_deepseek else settings.llm_model_name,
        generation_model_name=settings.generation_model_name,
        evaluation_model_name=settings.evaluation_model_name,
        collaboration_mode=collaboration_mode,
        embedding_model_name=settings.embedding_model_name,
        pipeline_mode=f"{settings.pipeline_mode} (skills={skill_status['skill_count']})",
        app_version=getattr(settings, "app_version", "V2.0"),
        app_update_date=getattr(settings, "app_update_date", "2026-05-24"),
        app_author=getattr(settings, "app_author", "ByteTitan-Star"),
        app_changelog=getattr(settings, "app_changelog", "引入 Harness 框架、ReAct Deep Search、用户系统、MySQL 持久化"),
    )